internal/controllers/valuations_controller.go

```diff
@@ -77,3 +89,84 @@ func (vc *ValuationsController) GetOffers(c *fiber.Ctx) error {
 
 	return c.JSON(dOffer)
 }
+
+// GetInstantOffer godoc
+// @Description gets instant offer for a particular user device
+// @Tags        user-devices
+// @Produce     json
+// @Success     200 {object} controllers.DeviceOffer
+// @Security    BearerAuth
+// @Router      /user/devices/{userDeviceID}/instant-offer [get]
+func (vc *ValuationsController) GetInstantOffer(c *fiber.Ctx) error {
+	udi := c.Params("userDeviceID")
+	userID := helpers.GetUserID(c)
+
+	ud, err := vc.userDeviceService.GetUserDevice(c.Context(), udi)
+
+	if err != nil {
+		return err
+	}
+
+	if ud.UserId != userID {
+		return fiber.NewError(fiber.StatusForbidden, "user does not have access to this vehicle")
+	}
+
+	wg := &sync.WaitGroup{}
+	m := &sync.RWMutex{}
+	ch := make(chan map[string]interface{})
+
+	wg.Add(1)
+	go func(ch chan map[string]interface{}, vin string, wg *sync.WaitGroup, m *sync.RWMutex) {
+		params := services.ValuationRequestData{}
+		offer, err := vc.drivlyService.GetOffersByVIN(*ud.Vin, &params)
+
+		if err != nil {
+			m.Lock()
+			ch <- map[string]interface{}{
+				"error": err,
+			}
+			m.Unlock()
+		}
+
+		if offer != nil {
+			m.Lock()
+			ch <- map[string]interface{}{
+				"offer": offer,
+			}
+			m.Unlock()
+		}
+
+		wg.Done()
+	}(ch, *ud.Vin, wg, m)
+
+	wg.Wait()
+
+	select {
+	case offer := <-ch:
+
+		dOffer := core.DeviceOffer{
+			OfferSets: []core.OfferSet{},
+		}
+
+		offerJSON, err := json.Marshal(offer["offer"].(map[string]interface{}))
+
+		if err != nil {
+			return err
+		}
+
+		ack, err := vc.natsService.JetStream.Publish(vc.natsService.OfferSubject, offerJSON)
```

Comment:
if have done the work to bring in NATS for async processing the offerJSON, why not just have all of this use NATS to process getting the offer from drivly & processing it. Should also help to get a quicker return on this endpoint / make it more reliable b/c we don't know how long drivly could take. 
The frontend can just pole the existing get offers endpoint to get the new offer asyncronously.